src/providers/oauth-provider.js

```diff
@@ -11,6 +11,7 @@ var defaults = {
   clientId: null,
   clientSecret: null,
   grantPath: '/oauth2/token',
+  grantType: 'password',
```

Comment:
I've been thinking about this and I figured out that there's an issue with simply adding the possibility to configure the grant_type field. `OAuth. getAccessToken` depends on `password` and `OAuth.getRefresh` has a different `grant_type` - `refresh_token`. I feel we can't merge this because it could possibly break the implementation. I think that what you want is related to #43.